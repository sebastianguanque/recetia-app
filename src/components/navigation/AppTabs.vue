<template>
  <nav class="tabs" role="tablist">
    <ul class="tabs-list">
      <li class="tab">
        <button
          id="recipe-creator-tab"
          :class="[
            'tab-button',
            { active: appStore.currentTab === 'recipe-creator' },
          ]"
          role="tab"
          :aria-selected="appStore.currentTab === 'recipe-creator'"
          aria-controls="recipe-creator"
          @click="appStore.switchTab('recipe-creator')"
        >
          <i class="fas fa-magic"></i> Creador de Recetas
        </button>
      </li>
      <li class="tab">
        <button
          id="shopping-list-tab"
          :class="[
            'tab-button',
            { active: appStore.currentTab === 'shopping-list' },
          ]"
          role="tab"
          :aria-selected="appStore.currentTab === 'shopping-list'"
          aria-controls="shopping-list"
          @click="appStore.switchTab('shopping-list')"
        >
          <i class="fas fa-shopping-cart"></i> Lista de Compras
        </button>
      </li>
      <li class="tab">
        <button
          class="tab-button"
          role="button"
          aria-label="Configuración de la aplicación"
          @click="showSettings"
        >
          <i class="fas fa-cog"></i> Configuración
        </button>
      </li>
    </ul>
  </nav>
</template>

<script setup>
import { useAppStore } from "@/stores/app";
import { useApiStore } from "@/stores/api";

const appStore = useAppStore();
const apiStore = useApiStore();

const showSettings = () => {
  const newKey = prompt("Ingresa nueva API Key de Gemini:", apiStore.apiKey);
  if (newKey && newKey.trim()) {
    apiStore.setApiKey(newKey.trim());
    alert("API Key actualizada correctamente");
  }
};
</script>

<style scoped>
.tabs {
  display: flex;
  background: #f8f9fa;
  border-bottom: 1px solid #e9ecef;
}

.tabs-list {
  display: flex;
  flex: 1;
  list-style: none;
  margin: 0;
  padding: 0;
}

.tab {
  display: flex;
  flex: 1;
}

.tab-button {
  flex: 1;
  padding: 15px 20px;
  border: none;
  background: transparent;
  cursor: pointer;
  font-size: 1rem;
  color: #666;
  transition: all 0.3s ease;
  border-bottom: 3px solid transparent;
}

.tab-button:hover {
  background: #e9ecef;
  color: #333;
}

.tab-button.active {
  color: #667eea;
  border-bottom-color: #667eea;
  background: white;
}

.tab-button i {
  margin-right: 8px;
}

@media (max-width: 768px) {
  .tabs-list {
    flex-direction: column;
  }
}
</style>
