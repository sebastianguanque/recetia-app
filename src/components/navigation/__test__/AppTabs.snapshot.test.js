// src/components/navigation/__tests__/AppTabs.snapshot.test.js
import { describe, it, expect, beforeEach } from "vitest";
import { mount } from "@vue/test-utils";
import { createPinia, setActivePinia, defineStore } from "pinia";
import AppTabs from "../AppTabs.vue";

describe("Snapshot visual: AppTabs.vue", () => {
  let pinia;
  let useAppStore;
  let useApiStore;
  let appStore;

  beforeEach(() => {
    pinia = createPinia();
    setActivePinia(pinia);

    useAppStore = defineStore("app", {
      state: () => ({
        currentTab: "recipe-creator",
      }),
      actions: {
        switchTab(tabName) {
          this.currentTab = tabName;
        },
      },
    });

    useApiStore = defineStore("api", {
      state: () => ({
        apiKey: "clave-inicial",
      }),
      actions: {
        setApiKey(newKey) {
          this.apiKey = newKey;
        },
      },
    });

    // Inicializamos las stores para asegurar estado
    appStore = useAppStore();
    useApiStore();
  });

  it("estructura inicial con pestaña 'recipe-creator' activa coincide con el snapshot", () => {
    const wrapper = mount(AppTabs, {
      global: { plugins: [pinia] },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("estructura con pestaña 'shopping-list' activa coincide con el snapshot", () => {
    appStore.switchTab("shopping-list");

    const wrapper = mount(AppTabs, {
      global: { plugins: [pinia] },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
