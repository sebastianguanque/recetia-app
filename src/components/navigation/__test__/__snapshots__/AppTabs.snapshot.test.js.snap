// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Snapshot visual: AppTabs.vue > estructura con pestaña 'shopping-list' activa coincide con el snapshot 1`] = `
"<nav data-v-642c808f="" class="tabs" role="tablist">
  <ul data-v-642c808f="" class="tabs-list">
    <li data-v-642c808f="" class="tab"><button data-v-642c808f="" id="recipe-creator-tab" class="tab-button" role="tab" aria-selected="false" aria-controls="recipe-creator"><i data-v-642c808f="" class="fas fa-magic"></i> Creador de Recetas </button></li>
    <li data-v-642c808f="" class="tab"><button data-v-642c808f="" id="shopping-list-tab" class="tab-button active" role="tab" aria-selected="true" aria-controls="shopping-list"><i data-v-642c808f="" class="fas fa-shopping-cart"></i> Lista de Compras </button></li>
    <li data-v-642c808f="" class="tab"><button data-v-642c808f="" class="tab-button" role="button" aria-label="Configuración de la aplicación"><i data-v-642c808f="" class="fas fa-cog"></i> Configuración </button></li>
  </ul>
</nav>"
`;

exports[`Snapshot visual: AppTabs.vue > estructura inicial coincide con el snapshot 1`] = `
"<nav data-v-642c808f="" class="tabs" role="tablist">
  <ul data-v-642c808f="" class="tabs-list">
    <li data-v-642c808f="" class="tab"><button data-v-642c808f="" id="recipe-creator-tab" class="tab-button active" role="tab" aria-selected="true" aria-controls="recipe-creator"><i data-v-642c808f="" class="fas fa-magic"></i> Creador de Recetas </button></li>
    <li data-v-642c808f="" class="tab"><button data-v-642c808f="" id="shopping-list-tab" class="tab-button" role="tab" aria-selected="false" aria-controls="shopping-list"><i data-v-642c808f="" class="fas fa-shopping-cart"></i> Lista de Compras </button></li>
    <li data-v-642c808f="" class="tab"><button data-v-642c808f="" class="tab-button" role="button" aria-label="Configuración de la aplicación"><i data-v-642c808f="" class="fas fa-cog"></i> Configuración </button></li>
  </ul>
</nav>"
`;

exports[`Snapshot visual: AppTabs.vue > estructura inicial con pestaña 'recipe-creator' activa coincide con el snapshot 1`] = `
"<nav data-v-642c808f="" class="tabs" role="tablist">
  <ul data-v-642c808f="" class="tabs-list">
    <li data-v-642c808f="" class="tab"><button data-v-642c808f="" id="recipe-creator-tab" class="tab-button active" role="tab" aria-selected="true" aria-controls="recipe-creator"><i data-v-642c808f="" class="fas fa-magic"></i> Creador de Recetas </button></li>
    <li data-v-642c808f="" class="tab"><button data-v-642c808f="" id="shopping-list-tab" class="tab-button" role="tab" aria-selected="false" aria-controls="shopping-list"><i data-v-642c808f="" class="fas fa-shopping-cart"></i> Lista de Compras </button></li>
    <li data-v-642c808f="" class="tab"><button data-v-642c808f="" class="tab-button" role="button" aria-label="Configuración de la aplicación"><i data-v-642c808f="" class="fas fa-cog"></i> Configuración </button></li>
  </ul>
</nav>"
`;
